src/marqo/tensor_search/tensor_search.py

```diff
@@ -563,9 +571,15 @@ def add_documents(config: Config, add_docs_params: AddDocsParams):
                             NsField.split_length]
                         split_overlap = index_info.index_settings[NsField.index_defaults][NsField.text_preprocessing][
                             NsField.split_overlap]
-                        content_chunks = text_processor.split_text(field_content, split_by=split_by,
+                        
+                        
+                        # text chunks: WITHOUT prefix, stored in backend chunk list
+                        text_chunks = text_processor.split_text(field_content, split_by=split_by,
                                                                    split_length=split_length, split_overlap=split_overlap)
-                        text_chunks = content_chunks
+                        
+                        # content chunks: WITH prefix, used to generate vectors (text prefix not actually stored in backend)
+                        content_chunks = text_processor.prefix_text_chunks(text_chunks, text_chunk_prefix)
```

Comment:
You could incorporate prefixing into splitting to avoid an extra pass over the text. Not sure how significant the difference will be though.

Reply:
I tried this, it's better that it's separate, because we want `text_chunks` to not have prefixes, while `content_chunks` have prefixes.